src/PermissionProvider/ConfigPermissions.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Drupal\entity_sync\PermissionProvider;
+
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+
+/**
+ * Provides permissions for each entity_sync.sync operation.
+ */
+class ConfigPermissions {
+
+  use StringTranslationTrait;
+
+  /**
+   * Returns an array of permissions for entity_sync.sync configs.
+   *
+   * @return array
+   *   The entity_sync.sync permissions.
+   *   @see \Drupal\user\PermissionHandlerInterface::getPermissions()
+   */
+  public function getPermissions() {
+    $permissions = [];
+
+    $config_factory = \Drupal::configFactory();
+
+    // Loop through each entity_sync.sync configurations and create
+    // corresponding permissions.
+    foreach (
+      $config_factory->listAll('entity_sync.sync.') as $config_name
+    ) {
+      $config = $config_factory->get($config_name)->get();
+
+      // Do not proceed if operations are not set for config.
+      if (!is_array($config['operations'])) {
+        continue;
+      }
+
+      $bundle = $config['entity']['bundle'];
+      $type_id = $config['entity']['type_id'];
+      $type_id_label = ucfirst($type_id);
```

Comment:
**CHANGE REQUEST**

The bundle is not required. We therefore need to build two types of permissions.
* If bundle available, call `buildEntityBundlePermissions` that will build permissions for syncs with entity type AND bundle.
* If bundle unavailable, call `buildEntityTypePermissions` that will build permissions for syncs with only entity type available.

**CHANGE REQUEST**

We need to validate that the enity type and bundle exist. Throw an `entity_sync\Exception\InvalidConfigurationException` if they don't.